### PR TITLE
Volcano plot

### PR DESCRIPTION
**Pull request by vincerubinetti**
- add volcano plot section
- add volcano data calculation of mean diff and p values for signatures, including welch's t test calculation referenced from another github issue
- add volcano plot visualization with axes and tooltips
- add util function to make transform strings easier to write
- add sample grouping to url persistence
- move all intensive synchronous operations (in this case, complex math computations) to `math.js`, and are imported as web workers, where needed, to be computed off the main thread
- split calculating enriched signatures into two steps: fetching/setting participations, and computing enriched signatures from the participations. this was previously done in one step, in the reducer, preventing the calculation from being offloaded to a web worker
- make default createAction like createFetchAction
- refactor controller.js a bit
- add useMounted hook for convenience
- change sample grouping from number based index to string based index

image

### CODE REVIEW

File: src/util/math.js [hunk 1]
```diff
@@ -87,16 +109,142 @@ export const calculateEnrichedSignatures = ({
...
+  }));
+
+  return volcanoSignatures;
+};
+
+// perform two-sample, unpaired, welch's (student's) t-test and return p value
+// from https://github.com/jstat/jstat/issues/189#issuecomment-469255296
+// note: cannot use the 'ttest' library because downstream dependency is
+// written in web assembly and cannot be compiled by 'workerize-loader' yet
+const ttest = (array1, array2) => {
```
Comment (dongbohu) [inline src/util/math.js:228]: The correctness of this function is critical, so we may need some tests to ensure that it will survive all edge cases. If you don't want to include the tests in this PR, can you add a new issue for now? Thanks.

**Review by dongbohu — COMMENTED** [review]
Left one comment.

File: src/util/math.js (hunk 1 above)
Comment (vincerubinetti) [reply src/util/math.js:228]: I agree. This, and pretty much everything else in `math.js` really need some rigorous testing and review from statistical people in the lab.

I think for now, I'll just spot check this against the previous interface, and then merge it. But I think the next big thing I want to do (perhaps before starting the Signatures page), it go back and get some real test coverage for the entire frontend. And that will include talking with people in the lab and getting some statistical test cases from them, comparing our code to other implementations (eg in python).

Comment (dongbohu) [reply src/util/math.js:228]: :+1:

Comment (vincerubinetti) [reply src/util/math.js:228]: Issue to track it:
https://github.com/greenelab/adage-frontend/issues/111

**Review by dongbohu — APPROVED** [review]
